File: projects/project 2/source/App1/App1/MainActivity.cs
using Android.App;
using Android.Widget;
using Android.OS;
using Android.Content;
using System.Collections.Generic;
using Android.Content.PM;
using Android.Provider;
//using App1;

namespace Camera_app
{
    [Activity(Label = "Camera_app", MainLauncher = true, Icon = "@mipmap/icon")]
    public class MainActivity : Activity
    {
        public static Java.IO.File _file;
        public static Java.IO.File _dir;

        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);

            // Set our view from the "main" layout resource
            SetContentView(Resource.Layout.Main);

            // Get our button from the layout resource,
            // and attach an event to it
            if (Canwetakeapic() == true)
            {
                Dir_for_pics();
                FindViewById<Button>(Resource.Id.button1).Click += Takingpic;
            }

        }

        private bool Canwetakeapic()
        {
            Intent intent = new Intent(MediaStore.ActionImageCapture);
            IList<ResolveInfo> availableActivities =
                PackageManager.QueryIntentActivities
                (intent, PackageInfoFlags.MatchDefaultOnly);
            return availableActivities != null && availableActivities.Count > 0;
        }

        private void Dir_for_pics()
        {
            _dir = new Java.IO.File(
                Android.OS.Environment.GetExternalStoragePublicDirectory(Android.OS.Environment.DirectoryPictures), "Picture");
            if (!_dir.Exists())
            {
                _dir.Mkdirs();
            }
        }

        private void Takingpic(object sender, System.EventArgs e)
        {
            Intent intent = new Intent(MediaStore.ActionImageCapture);
            _file = new Java.IO.File(_dir, string.Format("myPhoto_{0}.jpg", System.Guid.NewGuid()));
            //android.support.v4.content.FileProvider
            //getUriForFile(getContext(), "com.mydomain.fileprovider", newFile);
            //FileProvider.GetUriForFile

            //The line is a problem line for Android 7+ development
            //intent.PutExtra(MediaStore.ExtraOutput, Android.Net.Uri.FromFile(_file));
            StartActivityForResult(intent, 0);
        }

        protected override void OnActivityResult(int requestCode, Result resultCode, Intent data)
        {
            base.OnActivityResult(requestCode, resultCode, data);

            ImageView _thePic = FindViewById<ImageView>(Resource.Id.PictureTaken);
            //int height = Resource.
            int height = Resources.DisplayMetrics.HeightPixels;
            //int height = Resources.DisplayMetrics.HeightPixels;
            //int height = (Resource.DisplayMetrics as Android.Util.DisplayMetrics).HeightPixels;
            int width = _thePic.Height;

            Android.Graphics.Bitmap bitmap = (Android.Graphics.Bitmap)data.Extras.Get("data");
            Android.Graphics.Bitmap copyBitmap = bitmap.Copy(Android.Graphics.Bitmap.Config.Alpha8, true);

            for(int i = 0; i < copyBitmap.Width; i++)
            {
                for (int j = 0; j < copyBitmap.Height; j++)
                {
                    int p = copyBitmap.GetPixel(i, j);
                    Android.Graphics.Color c = new Android.Graphics.Color(p);
                    c.R = 0;
                    copyBitmap.SetPixel(i, j, c);
                }
            }
            if (bitmap != null)
            {
                _thePic.SetImageBitmap(bitmap);
                _thePic.Visibility = Android.Views.ViewStates.Visible;
                bitmap = null;
            }
            System.GC.Collect();
        }
    }
}

File: projects/project 2/JKP2/app/src/main/java/com/kinne/jack/jkp2/MainActivity.java
package com.kinne.jack.jkp2;
//Jack Daniel Kinne.  Project 2.  Mobile Apps CS 480.
import android.content.Intent;
import android.graphics.Bitmap;
import android.net.Uri;
import android.os.Environment;
import android.provider.MediaStore;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import java.io.File;
import java.text.SimpleDateFormat;
import java.util.Date;

public class MainActivity extends AppCompatActivity {

    //button for taking pictures
    Button btnpic;
    //button for transfering activities
    Button actChange;
    //image we will be messing with
    ImageView imgTakenPic;
    Bitmap bitmap;
    private static final int CAM_REQUEST=1313;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //taking a picture buton and event
        btnpic = (Button) findViewById(R.id.takePhotoButton);
        imgTakenPic = (ImageView)findViewById(R.id.imageView);
        btnpic.setOnClickListener(new btnTakePhotoClicker());

        //changing activity button and event
        actChange = (Button) findViewById(R.id.editPhotoButton);
        actChange.setOnClickListener(new changeActivity());
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        try {
            if (requestCode == CAM_REQUEST) {
                bitmap = (Bitmap) data.getExtras().get("data");
                imgTakenPic.setImageBitmap(bitmap);

            }
        }
        catch  (Exception e){}
    }

    //take a photo
    class btnTakePhotoClicker implements  Button.OnClickListener{

        @Override
        public void onClick(View view) {
            Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
            startActivityForResult(intent, CAM_REQUEST);
        }
    }

    //change activity
    class changeActivity implements  Button.OnClickListener{

        @Override
        public void onClick(View view) {
            Intent intent = new Intent(getApplicationContext(),Activity2.class);
            // to pass bitmap across!
            intent.putExtra("bitmap", bitmap );
            startActivity(intent);
        }
    }


}